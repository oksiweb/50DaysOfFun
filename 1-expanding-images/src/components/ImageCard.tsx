import React from "react"
import { Image } from "../types/image"

interface ImageCardProps extends Image {
  onClick?: (id: number) => void
}

function ImageCard({ id, imageUrl, title, active, onClick }: ImageCardProps) {
  const handleClick = () => {
    onClick?.(id)
  }

  return (
    <div className={`panel ${active && "active"}`} data-testid="image-card" onClick={handleClick} style={{ backgroundImage: `url(${imageUrl})` }}>
      <h3>{title}</h3>
    </div>
  )
}

export default ImageCard
