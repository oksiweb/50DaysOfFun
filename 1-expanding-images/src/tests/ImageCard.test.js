import React from "react"
import { render, fireEvent } from "@testing-library/react"
import ImageCard from "../components/ImageCard"

const image = {
  id: 1,
  imageUrl: "https://example.com/image.jpg",
  title: "Test Image",
  active: false
}

describe("ImageCard", () => {
  it("should render correctly", () => {
    const { getByText, getByTestId } = render(<ImageCard {...image} />)
    expect(getByTestId("image-card")).toBeInTheDocument()
    expect(getByText("Test Image")).toBeInTheDocument()
    expect(getByTestId("image-card")).toMatchSnapshot()
  })

  it("should add active class on click", () => {
    const handleClick = jest.fn()
    const { getByTestId } = render(<ImageCard {...image} active={true} onClick={handleClick} />)
    const imageCard = getByTestId("image-card")
    fireEvent.click(imageCard)
    expect(handleClick).toHaveBeenCalledWith(1)
    expect(imageCard).toHaveClass("active")
    expect(getByTestId("image-card")).toMatchSnapshot()
  })

  it("should not add active class if not active", () => {
    const handleClick = jest.fn()
    const { getByTestId } = render(<ImageCard {...image} onClick={handleClick} />)
    const imageCard = getByTestId("image-card")
    fireEvent.click(imageCard)
    expect(handleClick).toHaveBeenCalledWith(1)
    expect(imageCard).not.toHaveClass("active")
    expect(getByTestId("image-card")).toMatchSnapshot()
  })
})
