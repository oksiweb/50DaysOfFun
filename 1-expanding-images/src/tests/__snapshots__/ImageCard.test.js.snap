// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard should add active class on click 1`] = `
<div
  class="panel active"
  data-testid="image-card"
  style="background-image: url(https://example.com/image.jpg);"
>
  <h3>
    Test Image
  </h3>
</div>
`;

exports[`ImageCard should not add active class if not active 1`] = `
<div
  class="panel false"
  data-testid="image-card"
  style="background-image: url(https://example.com/image.jpg);"
>
  <h3>
    Test Image
  </h3>
</div>
`;

exports[`ImageCard should render correctly 1`] = `
<div
  class="panel false"
  data-testid="image-card"
  style="background-image: url(https://example.com/image.jpg);"
>
  <h3>
    Test Image
  </h3>
</div>
`;
